File: 2017/17/part2.py
# steps = 3
steps = 369

state = []
index = 1
for i in xrange(50000000):
  index = (index + 1 + steps) % len(state) if len(state) > 1 else 1
  # print('insert {} after position {}: {}'.format(i, index, state))
  state.insert(index, i)
  # print(index, state)

print(state[state.index(0) + 1])

File: 2017/17/part2.js
// const steps = 3;
const steps = 369;

let state = [0];
let index = 1;
for (let i = 1; i <= 50000000; ++i) {
  index = state.length > 1 ? (index + 1 + steps) % state.length : 1;
  // console.log(`insert ${i} after position ${index}: ${state}`);
  state.splice(index, 0, i);
  // console.log(index, state);
}

console.log(state[state.indexOf(0) + 1]);

